sourcegraph/sourcegraph
# web: move repo fetching errors to repo container

## Context

After improving the waterfall fetching for repo pages, we stopped showing the repo cloning progress page. 

1. This PR fixes this issue by consolidating all the repo fetching error messages in one place. 
2. This PR adds an integration test for the repo cloning UI. Screenshots of this page are available in [the Percy diff](https://percy.io/Sourcegraph/Sourcegraph/builds/21596169/changed/1204387093?browser=chrome&browser_ids=23&subcategories=approved&viewLayout=side-by-side&viewMode=new&width=1920&widths=1920).

Fixes #41731

## Test plan

1. Remove the repo on disk by deleting the entire directory:

```
rm -rf ~/.sourcegraph/repos/github.com/sourcegraph/sourcegraph
```

3. Run the dev server using `sg start`
4. Browse to the repo page: https://sourcegraph.test:3443/github.com/sourcegraph/sourcegraph
5. Ensure that the repo cloning UI is displaed.

## App preview:

- [Web](https://sg-web-vb-repo-container-errors.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-bpsdxigpwa.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
